commit 84922d331768ba9046d9737dbe66468ac45d41f2
Author: Cosimo Cecchi <cosimoc@gnome.org>
Date:   Mon Oct 15 18:45:49 2012 -0400

    appchooserwidget: check for g_app_info_should_show()
    
    This ensures NoDisplay=true or NotShowIn/OnlyShowIn values in the
    desktop file are respected.

diff --git a/gtk/gtkappchooserwidget.c b/gtk/gtkappchooserwidget.c
index ea4ac44..36875df 100644
--- a/gtk/gtkappchooserwidget.c
+++ b/gtk/gtkappchooserwidget.c
@@ -536,6 +536,9 @@ gtk_app_chooser_widget_add_section (GtkAppChooserWidget *self,
           !g_app_info_supports_files (app))
         continue;
 
+      if (!g_app_info_should_show (app))
+        continue;
+
       if (g_list_find_custom (exclude_apps, app,
                               (GCompareFunc) compare_apps_func))
         continue;
